src/main/java/com/google/sps/servlets/PersonServlet.java

```diff
@@ -38,36 +40,46 @@ public void init() {
     init(new DatastorePersonDao());
   }
 
-  // TODO: add a FakePersonDao class so this will become useful
   public void init(PersonDao personDao) {
     this.personDao = personDao;
   }
 
-  // Sends the request's contents to Datastore in the form of a new Person.
+  // Sends the request's contents to Datastore in the form of a new Person. Sends a 400 error if
+  // the JSON is malformed.
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    personDao.create(
-        Person.create(
-            request.getParameter("user-email"),
-            request.getParameter("first-name"),
-            request.getParameter("last-name"),
-            request.getParameter("company"),
-            request.getParameter("job"),
-            request.getParameter("linkedin")));
+    try {
+      PersonRequest personRequest =
+          new Gson().fromJson(getJsonString(request), PersonRequest.class);
+      personDao.create(Person.create(personRequest));
+    } catch (Exception JsonSyntaxException) {
+      response.sendError(400);
+      return;
+    }
   }
 
-  // Updates Datastore with the Person information in request.
+  // Updates Datastore with the Person information in request. Sends a 400 error if
+  // the JSON is malformed.
   @Override
   public void doPut(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    personDao.update(
-        Person.create(
-            request.getParameter("user-email"),
-            request.getParameter("first-name"),
-            request.getParameter("last-name"),
-            request.getParameter("company"),
-            request.getParameter("job"),
-            request.getParameter("linkedin")));
-    // TODO: redirect to home page and handle in JS.
+    try {
+      PersonRequest personRequest =
+          new Gson().fromJson(getJsonString(request), PersonRequest.class);
+      personDao.update(Person.create(personRequest));
+    } catch (Exception JsonSyntaxException) {
+      response.sendError(400);
+      return;
+    }
+  }
```

Comment:
All of these need to check authentication. As it is, any user could construct a request to modify any other user.